kernel/src/process.rs

```diff
@@ -91,6 +92,34 @@ struct LoadInfo {
     bss_size: u32,
     pkg_name_offset: u32,
     pkg_name_size: u32,
+    checksum: u32,
+}
+
+/// Converts a pointer to memory to a LoadInfo struct
+///
+/// This function takes a pointer to arbitrary memory and Optionally returns a
+/// LoadInfo struct. This function will validate the header checksum, but does
+/// not perform sanity or security checking on the structure
+unsafe fn parse_and_validate_load_info(address: *const u8) -> Option<&'static LoadInfo> {
+    let load_info = &*(address as *const LoadInfo);
+
+    if load_info.version != 1 {
+        return None;
+    }
+
+    let checksum =
+        load_info.version ^ load_info.total_size ^ load_info.entry_offset ^
```

Comment:
Do you not need to xor with 1 first here (as you did in `userland/tools/elf2tbf/src/main.rs` line 168)?